### PR TITLE
AK+Userland: Remove null state of DFS and raw usage of StringImpl

### PR DESCRIPTION
**Pull request by DanShaders**
Not sure how to comprehensively test all these libjs changes but I don't regress test262 and in-tree tests.

### CODE REVIEW

File: Userland/Libraries/LibJS/AST.h [hunk 1]
```diff
@@ -349,18 +349,17 @@ class ScopeNode : public Statement {
 
 // ImportEntry Record, https://tc39.es/ecma262/#table-importentry-record-fields
 struct ImportEntry {
-    DeprecatedFlyString import_name; // [[ImportName]] if a String
-    DeprecatedFlyString local_name;  // [[LocalName]]
-    bool is_namespace { false };     // [[ImportName]] if `namespace-object`
+    Optional<DeprecatedFlyString> import_name; // [[ImportName]] if a String
```
Comment (davidot) [inline Userland/Libraries/LibJS/AST.h:352]: Perhaps extend this comment with, present if a String, Empty if namespace-object

**Review by davidot — APPROVED** [review]
Looks good to me, I think that test262 should probably cover all the cases, so if there is no regression there than I would expect this to be fine.

**Comment by ADKaster** [comment]
Conflicts with the Json change, which has a similar commit in this PR (same commit? lol).

Have you run the test262 tests locally?

**Comment by DanShaders** [comment]
> Conflicts with the Json change, which has a similar commit in this PR (same commit? lol).

:facepalm:

**Review by ADKaster — APPROVED** [review]
Big fan of removing the null state from this guy. Hopefully we can get the rename to FlyBytes in at some point too.

**Pull request merged**